Fix was a part of merge of master:
e421d7b026e77f4dc127de49c01e80d533eb6825

--- ./gdb/symfile.c	2010-02-28 23:22:31.000000000 +0100
+++ ./gdb/symfile.c	2010-02-28 23:29:00.000000000 +0100
@@ -3448,6 +3448,7 @@ reread_symbols (void)
 	      objfile->symtabs = NULL;
 	      objfile->psymtabs = NULL;
 	      objfile->psymtabs_addrmap = NULL;
+	      objfile->quick_addrmap = NULL;
 	      objfile->free_psymtabs = NULL;
 	      objfile->cp_namespace_symtab = NULL;
 	      objfile->msymbols = NULL;
@@ -3458,6 +3459,8 @@ reread_symbols (void)
 	      memset (&objfile->msymbol_demangled_hash, 0,
 		      sizeof (objfile->msymbol_demangled_hash));
 
+	      objfile->flags &= ~OBJF_SYMTABS_READ;
+
 	      objfile->psymbol_cache = bcache_xmalloc ();
 	      objfile->macro_cache = bcache_xmalloc ();
 	      /* obstack_init also initializes the obstack so it is
@@ -3502,8 +3505,6 @@ reread_symbols (void)
 		  wrap_here ("");
 		}
 
-	      objfile->flags &= ~OBJF_SYMTABS_READ;
-
 	      /* We're done reading the symbol file; finish off complaints.  */
 	      clear_complaints (&symfile_complaints, 0, 1);
 
